lib/matplotlib/axis.py

```diff
@@ -1437,32 +1435,37 @@ def grid(self, b=None, which='major', **kwargs):
             Define the line properties of the grid, e.g.::
 
                 grid(color='r', linestyle='-', linewidth=2)
-
         """
-        if len(kwargs):
-            if not b and b is not None:  # something false-like but not None
```

Comment:
Shouldn't we instead rename `b` to `visible`?

Reply:
`b=None` (== toggle) and `visible=None` (== False) behave differently.  I think all these toggling APIs are pretty terrible, but breaking all the scripts in the world that do `ax.grid()` to toggle the grid on is just not worth it.

I guess we *could* aim to have the end API be `def grid(visible=True, ...):` (and deprecate `b`) though.  Then the only case we'd break is people doing `ax.grid()` to switch off the grid when the grid is already on, which is probably much less common (and `ax.grid(False)` just seems much clearer).  But let's keep that discussion for another time.

Reply:
Uh, it's worse than I thought.